### ISSUE TITLE
Routing requests to a different branch on the same domain

### ISSUE THREAD

**Issue by noelforte** [aws-amplify/amplify-console#511]
**Is your feature request related to a problem? Please describe.**
I was reviewing the other feature requests in this repository and came across #331 which I'd like to propose an extension to. Whether this is another feature or can be incorporated into A/B testing behavior, I think it would be beneficial to route requests to another feature branch deployment but serve it from the same domain so if you're creating a private area of a Amplify site or maybe some logged in version of the site, you can access it but preserve the URL.

**Describe the solution you'd like**
I'd like to be able to route traffic to a specific branch, kind of how Netlify does it currently with A/B testing as mentioned in #331 but by setting a cookie or something similar in order to keep the domain the same but view the other branch under different conditions (such as a logged-in state or something similar).

If this is already possible to accomplish with amplify right now, I'd be interested to know how!

**Comment by kahdojay** [comment]
Thanks for your feedback! I've added this to our backlog for further review by the team.

**Comment by kahdojay** [comment]
Hi @noelforte can you provide some more detail about how you'd expect to use this sort of feature and how it would be distinct from #331?

**Comment by noelforte** [comment]
Certainly, and the way that competitors have A/B testing set up allows for this sort of behavior but it's really more of a hack rather than an official feature. While password protecting an entire site with basic http-auth is good in the short term to keep a branch deploy private, the purposes of my suggestion is to allow branches to be routed to behind the scenes, say like if you were testing a private beta of an application or service, or if you had a state of a site that you only wanted to show to logged in users identified through either a local storage object or a cookie that contains information about what branch should be displayed. 

So, for my case, say I wanted to run a private beta on a site im building simultaneously with the  publically accessible version on the same domain. I'd like to be able to set a cookie or some way to "authorize" clients in order to divert them to my beta while keeping the URL the same so it looks like the same site. A/B testing would help if you're running a beta where you want to randomly separate traffic but I'd like to shape it in a more controlled way.

Hope that's clear enough, let me know if there's any other questions.

**Comment by rallona** [comment]
Are there any improvement related with split testing on amplify? I need it for my app!
